File: data_structure/base/9012.swift
let n = Int(readLine()!)!

for _ in 0..<n {
    let vps = readLine()!
    print(isVPS(vps) ? "YES" : "NO")
}

func isVPS(_ vps: String) -> Bool {
    var count = 0
    for c in vps {
        if c == "(" {
            count += 1
        } else if c == ")" {
            count -= 1
        }

        if count < 0 { return false }
    }
    return count == 0 ? true : false
}

// 괄호

File: javascript/data_structure/base/9012.js
// let [n, ...input] = require('fs').readFileSync('input.txt').toString().split('\n');
let [n, ...input] = require('fs').readFileSync('/dev/stdin').toString().split('\n');

function solution(ps) {
    let count = 0;
    
    for (const c of ps) {
        if (c === "(") {
            count += 1;
        } else if (c === ")") {
            count -= 1;
        }
        if (count < 0) { return "NO"; }
    }
    
    return count === 0 ? "YES" : "NO";
}

for (let i = 0; i < n; i++) {
    const ps = input[i].split("");
    console.log(solution(ps))
}
